src/vs/workbench/parts/extensions/electron-browser/extensionsViews.ts

```diff
@@ -240,7 +240,8 @@ export class ExtensionsListView extends ViewletPanel {
 
 		if (names.length) {
 			return this.extensionsWorkbenchService.queryGallery({ names, source: 'queryById' })
-				.then(pager => new PagedModel(pager));
+				.then(pager => new PagedModel(pager),
```

Comment:
Are you sure this is the part where the error is not bubbling up? The absence of error callback in a `then` would mean that the rejected promise would bubble up. I believe the problem is when we await of the `queryGalley` calls which occurs quite a few times in the `query` method.

A few ways to fix this
- Add try/catch around all the awaits on queryGallery 
- Convert the await to normal promise. So 
```
const pager = await this.extensionsWorkbenchService.queryGallery(options);
		return new PagedModel(pager);
```
would become
```
return this.extensionsWorkbenchService.queryGallery(options).then(pager => new PagedModel(pager));
```